### ISSUE TITLE
Is it a problem that events pushed from the server aren't acked by the client?

### ISSUE THREAD

**Issue by ara4n** [matrix-org/matrix-websockets#5]
/sync provides an implicit ACK that events have been received by the client, but I think we lose that here?

**Comment by richvdh** [comment]
Yes, this is a problem now that we treat an incremental sync as an ack of to-device events in the previous sync. 

We're going to need to add acks to the websockets protocol, and provide a modified sync api for the proxy to talk to the HS over.
